src/main/java/com/laserfiche/api/client/APIServer/TokenClientImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.laserfiche.api.client.APIServer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.laserfiche.api.client.model.ApiException;
+import com.laserfiche.api.client.model.CreateConnectionRequest;
+import com.laserfiche.api.client.model.OffsetDateTimeDeserializer;
+import com.laserfiche.api.client.model.ProblemDetails;
+import com.laserfiche.api.client.oauth.TokenClientObjectMapper;
+import kong.unirest.Header;
+import kong.unirest.HttpResponse;
+import kong.unirest.Unirest;
+import kong.unirest.json.JSONObject;
+import org.threeten.bp.OffsetDateTime;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class TokenClientImpl implements TokenClient {
+    private String baseUrl;
+
+    protected ObjectMapper objectMapper;
+
+    public TokenClientImpl(String baseUrl) {
+        this.baseUrl = baseUrl;
+        Unirest
+                .config()
+                .setObjectMapper(new TokenClientObjectMapper());
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(OffsetDateTime.class, new OffsetDateTimeDeserializer());
+        this.objectMapper = JsonMapper
+                .builder()
+                .addModule(module)
+                .disable(MapperFeature.CAN_OVERRIDE_ACCESS_MODIFIERS)
+                .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+                .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
+                .build();
+    }
+
+    @Override
+    public CompletableFuture<SessionKeyInfo> createAccessToken(String repoId, CreateConnectionRequest body) {
```

Comment:
Why do we call `CreateConnectionRequest` body?

Reply:
just keeping it consistent with one the C# and how it is named in the swagger.json file for selfhosted